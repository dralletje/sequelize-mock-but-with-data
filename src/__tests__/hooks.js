// @ts-nocheck
let Sequelize = require("sequelize");

// I should move to observatory or something,
// but this works for now
const DATE_TO_USE = new Date("2016");
const _Date = Date;
global.Date = jest.fn(() => DATE_TO_USE);
global.Date.UTC = _Date.UTC;
global.Date.parse = _Date.parse;
global.Date.now = _Date.now;

let define = (options = {}) => {
  let sequelize = new Sequelize();
  let User = sequelize.define(
    "User",
    {
      id: {
        type: Sequelize.UUID,
        defaultValue: Sequelize.UUIDV4,
      },
      name: Sequelize.STRING,
      email: Sequelize.STRING,
      age: Sequelize.INTEGER,
      verified: Sequelize.BOOLEAN,
    },
    options
  );
  return { sequelize, User };
};

let some_users_setup = async ({ User }) => {
  await User.create({
    name: "Michiel Dral",
    email: "m.c.dral@gmail.com",
    age: 22,
    verified: true,
  });
  await User.create({
    name: "Ola Beige",
    email: "o.l.beige@gmail.com",
    age: 22,
    verified: true,
  });
  await User.create({
    name: "Jake Strange",
    email: "j.strange@gmail.com",
    age: 24,
    verified: true,
  });
};

it("should work with simple insert", async () => {
  let { User } = define();

  let afterFindHook = jest.fn();
  User.addHook("afterFind", (items) => {
    afterFindHook();
    for (let item of items) {
      item.canDrink = item.age > 18;
      delete item.verified;
    }
  });

  some_users_setup({ User });

  let user = await User.findOne({
    where: {
      email: "m.c.dral@gmail.com",
    },
  });

  expect(afterFindHook).toBeCalled();
  expect(user).toMatchSnapshot();
});
