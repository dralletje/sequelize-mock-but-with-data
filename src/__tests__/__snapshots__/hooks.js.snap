// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work with simple insert 1`] = `
Object {
  "__table_name__": "User",
  "age": 22,
  "canDrink": true,
  "createdAt": 2016-01-01T00:00:00.000Z,
  "email": "m.c.dral@gmail.com",
  "id": "1",
  "name": "Michiel Dral",
  "updatedAt": 2016-01-01T00:00:00.000Z,
}
`;
